### PR TITLE
[IMPROVEMENT] Do not run the CI if it's a draft pull request

### PR DESCRIPTION
**Pull request by ArchitBhonsle**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

A simple two line change that also checks whether the pull request under question is not a draft request.

### CODE REVIEW

**Comment by canihavesomecoffee** [comment]
Can you add a unit test for it as well? :)

**Comment by canihavesomecoffee** [comment]
@ArchitBhonsle can you check why it fails to apply pydocstyle?

**Comment by ArchitBhonsle** [comment]
Sorry for the ugly force push. It had been a long time since I worked on this and the progress wasn't that significant so I just nuked it.

**Comment by ArchitBhonsle** [comment]
I decided to simply `pass` draft PRs. This preserves the original flow of dumping `{'msg': 'EOL'}`. Is that OK or should it return something else?

https://github.com/ArchitBhonsle/sample-platform/blob/43cf1bb96cdf5555a294a9be0c7ee9552e9af42a/mod_ci/controllers.py#L890-

**Comment by ArchitBhonsle** [comment]
Oh never mind. This implementation has a bug. It checks whether a PR is draft first. This causes an issue for a **closed draft**. The running tasks are not cancelled as the function never reaches the "closed" `elif`.

**Review by thealphadollar — COMMENTED** [review]
The PR looks good to me. Thanks for the contribution, Archit.

Let's add/update unit tests and then we can merge it 🚢

**Review by thealphadollar — CHANGES_REQUESTED** [review]
Hey! This looks good to me. Would you mind adding few new tests for draft:true PRs?

**Comment by ArchitBhonsle** [comment]
Yup I was just looking at the WebHook documentation and checking if I'm missing anything. Looks like I am. I think `ready_for_review` should also trigger the CI.

**Review by thealphadollar — APPROVED** [review]
LGTM! Thanks for adding the tests, Archit 🚢 

Let us see why the tests are failing.

**Comment by ArchitBhonsle** [comment]
Yup I understand why my tests are failing. I didn't add the arguments passed by the decorators. But not sure why the other tests are. I'll look into it.

**Comment by Tarun-Arora** [comment]
@ArchitBhonsle we should also stop the tests if possible when the PR is just marked as start.
You may also refer to PR #792 for this and unittests as well. And we can merge these PRs.

**Comment by ArchitBhonsle** [comment]
@Tarun-Arora that would be when the PR is `converted_to_draft` right?

**Comment by ArchitBhonsle** [comment]
The tests were failing because I added `draft` at the wrong level. `payload->draft` instead of `payload->pull_request->draft` 🫠

**Review by canihavesomecoffee — APPROVED** [review]

**Comment by canihavesomecoffee** [comment]
Nice work, thanks :+1:

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -902,7 +908,7 @@ def start_ci():
                 if repository.get_pull(number=pr_nr).mergeable is not False:
                     add_test_entry(g.db, commit_hash, TestType.pull_request, pr_nr=pr_nr)
 
-            elif payload['action'] == 'closed':
+            elif inactive:
                 g.log.debug('PR was closed, no after hash available')
                 # Cancel running queue
                 tests = Test.query.filter(Test.pr_nr == pr_nr).all()
```
Comment (Tarun-Arora) [inline mod_ci/controllers.py:914]: @ArchitBhonsle We should also update the message on line 919. "PR closed", "PR marked as draft".

**Pull request merged**